_posts/2017-04-17-stackstorm-architecture-components.markdown

```diff
@@ -166,3 +166,117 @@ The specific application being run on Python is [Gunicorn](http://gunicorn.org/)
 
 That's about it! `st2api` is actually remarkably simple once you understand how the Swagger spec resolves to actual Python functions (again please read (TODO insert link to post here) for more info on that portion) and how the WSGI server is instantiated.
 
+
+### Mistral
+
+[Mistral](https://docs.stackstorm.com/mistral.html) is very important because it is StackStorm's primary workflow engine. If you're unfamiliar with [Mistral](https://wiki.openstack.org/wiki/Mistral), it is an OpenStack project for managing workflows. Since Mistral provides a lot of useful workflow primitives, StackStorm integrates with Mistral, and provides plugins to allow Mistral to execute StackStorm actions.
+
+Similar to StackStorm itself, we're not just talking about one component when we're talking about Mistral. Mistral itself is made up of two separate running processes:
+
+- **`mistral-api`** allows 3rd-party systems to execute workflows, get execution status, etc.
+- **`mistral-server`** schedules and runs workflows and tasks
+
+Before looking at how Mistral and StackStorm interoperate, let's spend some time diving into these components, and understanding Mistral's architecture on it's own. [The Mistral Architectural Overview](https://docs.openstack.org/developer/mistral/architecture.html) page is a very useful place to start understanding this. Please review at least this page before continuing, as we'll be referring to certain components later.
+
+If you're just getting started with Mistral and want to explore, you should start looking at the `mistral` tool. This is the primary mechanism for CLI interactions with Mistral, and it's one of the best ways to troubleshoot or just explore how Mistral works. As with many CLI tools, `mistral -h` shows the commands you can run and a description of each. If you're familiar with the StackStorm CLI client `st2`, much of the terminology will be familiar to you.
```

Comment:
I would make this paragraph less relevant. I prefer user to use Mistral from StackStorm. The mistral CLI is only used for troubleshooting purposes and I don't recommend asking user to start exploring from there.  They should start exploring by running workflow directly from StackStorm.